spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteTablePathSparkAction.java

```diff
@@ -572,21 +564,12 @@ private void rewritePositionDeletes(TableMetadata metadata, Set<DeleteFile> toRe
     Dataset<DeleteFile> deleteFileDs =
         spark().createDataset(Lists.newArrayList(toRewrite), deleteFileEncoder);
 
-    Broadcast<Table> serializableTable = sparkContext().broadcast(SerializableTable.copyOf(table));
-    Broadcast<Map<Integer, PartitionSpec>> specsById =
-        sparkContext().broadcast(metadata.specsById());
-
     PositionDeleteReaderWriter posDeleteReaderWriter = new SparkPositionDeleteReaderWriter();
     deleteFileDs
         .repartition(toRewrite.size())
         .foreach(
             rewritePositionDelete(
-                serializableTable,
-                specsById,
-                sourcePrefix,
-                targetPrefix,
-                stagingDir,
-                posDeleteReaderWriter));
+                tableBroadcast(), sourcePrefix, targetPrefix, stagingDir, posDeleteReaderWriter));
```

Comment:
rewritePositionDelete is a private helper function, can we just call tableBroadcast() in that function rather than have it be an argument. This will also help address the whole private field masking issue